### ISSUE TITLE
Bugs & Suggestions

### ISSUE THREAD

**Issue by yfdyh000** [iainbeeston/tab-less#16]
1. Can't see the icon for browserAction in Firefox, looks `"default_icon"` is bad in manifest.json.
2. Customize mode looks broken in Firefox.
3. No updateBadge if just started.
4. Close other windows not work for me, due to `state: "maximized"`.
5. Add "Close other windows" to tabs context menu.
6. Provide a way (like userChrome.css) hides the tabs bar, people will like it.
7. Set your LGPL license to GitHub repo. like [this](https://stackoverflow.com/questions/31639059/how-to-add-license-to-an-existing-github-project).

**Comment by iainbeeston** [comment]
Thanks for your feedback, many of those are things that I hadn't noticed before. I'll answer your points in order:

1. I'm not sure what is causing this - the icon looks correct for me in Firefox 62 on Mac. However, I've re-read Mozilla's icon guidelines, and I realised that Firefox uses different icon sizes to Google Chrome. I've added those missing icon sizes now, so hopefully the icon will appear correctly now (b002775e6a8eaa325e9805f26a75469859294def)
2. Unfortunately this is a bug in Firefox itself and I haven't been able to find a workaround 😢  https://bugzilla.mozilla.org/show_bug.cgi?id=1494596
3. I'm not sure what is causing this. However, I've updated the code in a way that I hope might fix the issue (3f9f23609b492839fc8f3a7c2529d9b9f4f1d534)
4. I changed the behaviour recently so that it does not close maximised or full screen windows, however I think this is confusing for a lot of people, so I have reverted that change, and now maximised windows will close as well (a7bfab9170a8d6feabaa4ac28fc438dd3f30103a)
5. That is a good idea. However, one of my goals with this browser extension is that it should request no additional permissions from the user (for example, many browser extensions request access to view the current web page), so that it is secure and respects the user's privacy. Unfortunately I cannot add context menus without requesting the context menus permission. So for now, I will not add that feature.
6. That is also a good idea, however I'm not sure how I would make that work in a cross platform way, and I worry that it could produce strange bugs as the browser UI changes over time (I might look into this in the future)
7. Good point, I hadn't noticed that github wasn't recognising the license (2953c6d007e92f2f104a7d3ad8f0c3364018a160)

I've released fixes for 1, 3 and 4 in version 0.6.0, which should now be available for Firefox and Chrome
